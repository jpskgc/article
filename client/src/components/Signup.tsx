import React from 'react';
import {Button, Form, Grid, Header, Message, Segment} from 'semantic-ui-react';
import firebase from '../firebase/firebase';
import {Redirect} from 'react-router';

interface userState {
  email: string;
  password: string;
  redirect: boolean;
}

class SignupForm extends React.Component<{}, userState> {
  constructor(props: {}) {
    super(props);
    this.state = {
      email: '',
      password: '',
      redirect: false,
    };
    this.signUp = this.signUp.bind(this);
    this.handleChangeEmail = this.handleChangeEmail.bind(this);
    this.handleChangePassword = this.handleChangePassword.bind(this);
    this.renderRedirect = this.renderRedirect.bind(this);
  }

  handleChangeEmail(e: React.FormEvent<HTMLInputElement>) {
    this.setState({email: e.currentTarget.value});
  }

  handleChangePassword(e: React.FormEvent<HTMLInputElement>) {
    this.setState({password: e.currentTarget.value});
  }

  signUp() {
    firebase
      .auth()
      .createUserWithEmailAndPassword(this.state.email, this.state.password)
      .then(res => {
        console.log('Create account: ', res.user);
      })
      .catch(error => {
        console.log(error.message);
      });
    this.setState({
      redirect: true,
    });
  }

  renderRedirect = () => {
    if (this.state.redirect) {
      return <Redirect to="" />;
    }
  };

  render() {
    return (
      <Grid textAlign="center" style={{height: '100vh'}} verticalAlign="middle">
        <Grid.Column style={{maxWidth: 450}}>
          <Header as="h2" color="teal" textAlign="center">
            Sign-up to your account
          </Header>
          <Form size="large" onSubmit={this.signUp}>
            <Segment stacked>
              <Form.Input
                fluid
                icon="user"
                iconPosition="left"
                placeholder="E-mail address"
                value={this.state.email}
                onChange={this.handleChangeEmail}
              />
              <Form.Input
                fluid
                icon="lock"
                iconPosition="left"
                placeholder="Password"
                type="password"
                value={this.state.password}
                onChange={this.handleChangePassword}
              />
              {this.renderRedirect()}
              <Button color="teal" fluid size="large">
                Signup
              </Button>
            </Segment>
          </Form>
          <Message>
            already have account? <a href="/login">Login</a>
          </Message>
        </Grid.Column>
      </Grid>
    );
  }
}

export default SignupForm;
