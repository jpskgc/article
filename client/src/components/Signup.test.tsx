import React from 'react';
import Signup from './Signup';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<Signup />).toJSON();
  expect(tree).toMatchSnapshot();
});
