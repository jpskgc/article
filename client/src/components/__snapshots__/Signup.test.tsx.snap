// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui center aligned middle aligned grid"
  style={
    Object {
      "height": "100vh",
    }
  }
>
  <div
    className="column"
    style={
      Object {
        "maxWidth": 450,
      }
    }
  >
    <h2
      className="ui teal center aligned header"
    >
      Sign-up to your account
    </h2>
    <form
      className="ui large form"
      onSubmit={[Function]}
    >
      <div
        className="ui stacked segment"
      >
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              onChange={[Function]}
              placeholder="E-mail address"
              type="text"
              value=""
            />
            <i
              aria-hidden="true"
              className="user icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <div
          className="field"
        >
          <div
            className="ui fluid left icon input"
          >
            <input
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
            />
            <i
              aria-hidden="true"
              className="lock icon"
              onClick={[Function]}
            />
          </div>
        </div>
        <button
          className="ui teal large fluid button"
          onClick={[Function]}
        >
          Signup
        </button>
      </div>
    </form>
    <div
      className="ui message"
    >
      already have account? 
      <a
        href="/login"
      >
        Login
      </a>
    </div>
  </div>
</div>
`;
